### PR TITLE
✨ Support multiple loggers

### PR DESCRIPTION
**Pull request by fuzzybinary**
### What and why?

Allow multiple loggers, create a default logger when a configuration is passed in initially, and allow clients to create loggers after initialization.

### Review checklist

- [x] This pull request has appropriate unit and / or integration tests 
- [x] This pull request references a Github or JIRA issue

### CODE REVIEW

**Review by xgouchet — APPROVED** [review]

File: packages/datadog_flutter_plugin/android/src/main/kotlin/com/datadoghq/flutter/DatadogSdkPlugin.kt [hunk 1]
```diff
@@ -102,10 +103,9 @@ class DatadogSdkPlugin : FlutterPlugin, MethodCallHandler {
             config.trackingConsent
         )
 
-        if (config.loggingConfiguration != null) {
-            logsPlugin = DatadogLogsPlugin()
-            logsPlugin?.setup(binding, config.loggingConfiguration!!)
-        }
+        // Always setup logging as a user can create a log after initialization
+        logsPlugin = DatadogLogsPlugin()
+        logsPlugin?.setup(binding)
```
Comment (0xnm) [inline packages/datadog_flutter_plugin/android/src/main/kotlin/com/datadoghq/flutter/DatadogSdkPlugin.kt:108]:
```suggestion
        logsPlugin = DatadogLogsPlugin().apply { setup(binding) }
```

File: packages/datadog_flutter_plugin/android/src/main/kotlin/com/datadoghq/flutter/DatadogLogsPlugin.kt [hunk 2]
```diff
@@ -26,119 +26,150 @@ class DatadogLogsPlugin : MethodChannel.MethodCallHandler {
...
-            )
+            else -> {
+                result.notImplemented()
+            }
+        }
+    }
+
+    private fun createLogger(loggerHandle: String, configuration: LoggingConfiguration) {
+        val logBuilder = Logger.Builder()
+            .setDatadogLogsEnabled(true)
```
Comment (0xnm) [inline packages/datadog_flutter_plugin/android/src/main/kotlin/com/datadoghq/flutter/DatadogLogsPlugin.kt:162]: shouldn't we read `configuration.sendLogsToDatadog` here? Although it is useful only for SDK development probably.

File: packages/datadog_flutter_plugin/ios/Classes/DatadogLogsPlugin.swift [hunk 3]
```diff
@@ -12,24 +12,31 @@ public class DatadogLogsPlugin: NSObject, FlutterPlugin {
...
-  func initialize(withLogger logger: Logger) {
-    self.logger = logger
+  func addLogger(logger: Logger, withHandle handle: String) {
+    loggerRegistry[handle] = logger
   }
 
-  func initialize(configuration: DatadogFlutterConfiguration.LoggingConfiguration) {
+  func createLogger(loggerHandle: String, configuration: DatadogLoggingConfiguration) {
     let builder = Logger.builder
       .sendNetworkInfo(configuration.sendNetworkInfo)
```
Comment (0xnm) [inline packages/datadog_flutter_plugin/ios/Classes/DatadogLogsPlugin.swift:27]: and I guess iOS SDK has no method for `sendLogsToDatadog` option in a builder?

File: packages/datadog_flutter_plugin/android/src/main/kotlin/com/datadoghq/flutter/DatadogSdkPlugin.kt (hunk 1 above)
Comment (fuzzybinary) [reply packages/datadog_flutter_plugin/android/src/main/kotlin/com/datadoghq/flutter/DatadogSdkPlugin.kt:108]: Should that pattern be carried down into the other setup calls?

Comment (0xnm) [reply packages/datadog_flutter_plugin/android/src/main/kotlin/com/datadoghq/flutter/DatadogSdkPlugin.kt:108]: As you wish :) It is a nitpick to me and doesn't really matter in the single-threaded environment.

File: packages/datadog_flutter_plugin/android/src/main/kotlin/com/datadoghq/flutter/DatadogLogsPlugin.kt (hunk 2 above)
Comment (fuzzybinary) [reply packages/datadog_flutter_plugin/android/src/main/kotlin/com/datadoghq/flutter/DatadogLogsPlugin.kt:162]: Yes, good catch.

File: packages/datadog_flutter_plugin/ios/Classes/DatadogLogsPlugin.swift (hunk 3 above)
Comment (fuzzybinary) [reply packages/datadog_flutter_plugin/ios/Classes/DatadogLogsPlugin.swift:27]: It does. Also a good catch :)

**Review by 0xnm — APPROVED** [review]

**Pull request merged**